### PR TITLE
Refactor `LeaseAcquirer` to be able to consume leaseId

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Acquire lease on blobs before deleting them](https://tools.hmcts.net/jira/browse/BPS-1205)

### Change description ###

First step - prepare the success function to accept leaseId in order to be able to use in `ContainerCleaner`. Reference: #544 - sort of complete working code

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
Realised. Had to do a spike first, but the approach was hacky/develop mode. This one starts to introduce proper historical reasoning

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java [hunk 1]
```diff
@@ -41,10 +43,22 @@ public void ifAcquiredOrElse(BlobClient blobClient, Runnable onSuccess, Runnable
...
+                    blobClient.getBlobName(),
+                    blobClient.getContainerName()
+                );
+            }
         }
     }
 
+    public void ifAcquiredOrElse(BlobClient blobClient, Runnable onSuccess) {
+        ifAcquiredOrElse(blobClient, leaseId -> onSuccess.run(), () -> {});
+    }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java:60]: 'OrElse' but there's no 'else'

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java [hunk 2]
```diff
@@ -22,12 +24,12 @@ public LeaseAcquirer(LeaseClientProvider leaseClientProvider) {
         this.leaseClientProvider = leaseClientProvider;
     }
 
-    public void ifAcquiredOrElse(BlobClient blobClient, Runnable onSuccess, Runnable onFailure) {
+    public void ifAcquiredOrElse(BlobClient blobClient, Consumer<String> onSuccess, Runnable onBlobNotFound) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java:27]: add javadoc on what the `onSuccess` takes as param

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java [hunk 3]
```diff
@@ -41,10 +43,22 @@ public void ifAcquiredOrElse(BlobClient blobClient, Runnable onSuccess, Runnable
...
-            onFailure.run();
+            if (exc.getErrorCode() == BLOB_NOT_FOUND) {
+                onBlobNotFound.run();
+            } else {
+                logger.info(
+                    "Cannot acquire a lease for blob - skipping. File name: {}, container: {}",
+                    blobClient.getBlobName(),
+                    blobClient.getContainerName()
+                );
+            }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java:54]: Should this change be in this PR?
It mentions enabling using lease id.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java:54]: it also mentions `Refactor`. It is basic look over. For following will have to handle `BLOB_NOT_FOUND` individually so ending up with 3 functional interface params in method. My gut feeling told team would not like it hence refactor a bit more. Very minimal tbh

**Pull request merged**